File: src/main/java/me/redth/armorochud/config/ModConfig.java
package me.redth.armorochud.config;

import cc.polyfrost.oneconfig.config.Config;
import cc.polyfrost.oneconfig.config.annotations.Color;
import cc.polyfrost.oneconfig.config.annotations.Dropdown;
import cc.polyfrost.oneconfig.config.annotations.HUD;
import cc.polyfrost.oneconfig.config.annotations.Switch;
import cc.polyfrost.oneconfig.config.core.OneColor;
import cc.polyfrost.oneconfig.config.data.Mod;
import cc.polyfrost.oneconfig.config.data.ModType;
import me.redth.armorochud.ArmorOCHUD;
import me.redth.armorochud.hud.ArrowHUD;
import me.redth.armorochud.hud.BootsHUD;
import me.redth.armorochud.hud.ChestplateHUD;
import me.redth.armorochud.hud.HelmetHUD;
import me.redth.armorochud.hud.LeggingsHUD;
import me.redth.armorochud.hud.HandHUD;

@SuppressWarnings("unused")
public class ModConfig extends Config {

    @Dropdown(name = "Durability Format", options = {"Percentage", "Remaining", "Remaining/Max"}, size = 2)
    public static int durabilityFormat = 2;

    @Switch(name = "Use Static Text Color")
    public static boolean useStaticTextColor = false;

    @Color(name = "Static Text Color")
    public static OneColor staticTextColor = new OneColor(0xFFFFFFFF);

    @HUD(name = "Hand", category = "Hand")
    public HandHUD handHUD = new HandHUD();

    @HUD(name = "Helmet", category = "Helmet")
    public HelmetHUD helmetHUD = new HelmetHUD();

    @HUD(name = "Chestplate", category = "Chestplate")
    public ChestplateHUD chestplateHUD = new ChestplateHUD();

    @HUD(name = "Leggings", category = "Leggings")
    public LeggingsHUD leggingsHUD = new LeggingsHUD();

    @HUD(name = "Boots", category = "Boots")
    public BootsHUD bootsHUD = new BootsHUD();

    @HUD(name = "Arrows", category = "Arrows")
    public ArrowHUD arrowHUD = new ArrowHUD();

    public ModConfig() {
        super(new Mod(ArmorOCHUD.NAME, ModType.HUD), ArmorOCHUD.MODID + ".json");
        initialize();
        addDependency("staticTextColor", "useStaticTextColor");
    }

}

File: src/main/kotlin/me/redth/armorochud/config/ModConfig.kt
package me.redth.armorochud.config

import cc.polyfrost.oneconfig.config.Config
import cc.polyfrost.oneconfig.config.annotations.Color
import cc.polyfrost.oneconfig.config.annotations.Dropdown
import cc.polyfrost.oneconfig.config.annotations.HUD
import cc.polyfrost.oneconfig.config.annotations.Switch
import cc.polyfrost.oneconfig.config.core.OneColor
import cc.polyfrost.oneconfig.config.data.Mod
import cc.polyfrost.oneconfig.config.data.ModType
import me.redth.armorochud.ArmorOCHUD
import me.redth.armorochud.hud.*

object ModConfig : Config(Mod(ArmorOCHUD.NAME, ModType.HUD), "${ArmorOCHUD.MODID}.json") {
    @Dropdown(name = "Durability Format", options = ["Percentage%", "Remaining", "Remaining/Max"], size = 2)
    var durabilityFormat = 2

    @Switch(name = "Use Static Text Color")
    var useStaticTextColor = false

    @Color(name = "Static Text Color")
    var staticTextColor = OneColor(0xFFFFFFFF.toInt())

    @HUD(name = "Hand", category = "Hand")
    var handHUD = HandHUD

    @HUD(name = "Helmet", category = "Helmet")
    var helmetHUD = HelmetHUD

    @HUD(name = "Chestplate", category = "Chestplate")
    var chestplateHUD = ChestplateHUD

    @HUD(name = "Leggings", category = "Leggings")
    var leggingsHUD = LeggingsHUD

    @HUD(name = "Boots", category = "Boots")
    var bootsHUD = BootsHUD

    @HUD(name = "Arrow", category = "Arrow")
    var arrowHUD = ArrowHUD

    init {
        initialize()
        addDependency("staticTextColor", "useStaticTextColor")
    }

}